libevm/legacy/legacy_test.go

```diff
@@ -0,0 +1,97 @@
+package legacy
+
+import (
+	"errors"
+	"testing"
+
+	"github.com/ava-labs/libevm/core/vm"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestPrecompiledStatefulContract_Upgrade(t *testing.T) {
+	t.Parallel()
+
+	testCases := map[string]struct {
+		env           *stubPrecompileEnvironment
+		input         []byte
+		cRet          []byte
+		cRemainingGas uint64
+		cErr          error
+		wantRet       []byte
+		wantErr       string
+		wantEnv       *stubPrecompileEnvironment
+	}{
+		"call_error": {
+			env: &stubPrecompileEnvironment{
+				gasToReturn: 10,
+			},
+			input:         []byte{1},
+			cRet:          []byte{2},
+			cRemainingGas: 6,
+			cErr:          errors.New("test error"),
+			wantRet:       []byte{2},
```

Comment:
Call me a maniac 😄 - but shouldn't we enforce to return a `nil` ret if the `c` call fails with an error?